lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,308 @@
+#[derive(Clone)]
+pub struct PhysicsProperties {
+    data : Rc<RefCell<PhysicsPropertiesData>>
+}
+
+impl PhysicsProperties {
+    pub fn new
+    (kinematics: KinematicsProperties, spring:SpringProperties, drag:DragProperties) -> Self {
+        let data = Rc::new(RefCell::new(PhysicsPropertiesData::new(kinematics, spring, drag)));
+        Self { data }
+    }
+}
+
+
+// === Getters ===
+
+impl PhysicsProperties {
+    pub fn kinematics(&self) -> KinematicsProperties { self.data.borrow().kinematics }
+    pub fn spring    (&self) -> SpringProperties     { self.data.borrow().spring }
+    pub fn drag      (&self) -> DragProperties       { self.data.borrow().drag }
+}
+
+
+// === Setters ===
+
+impl PhysicsProperties {
+    pub fn mod_kinematics<F:FnOnce(&mut KinematicsProperties)>(&mut self, f:F) {
+        f(&mut self.data.borrow_mut().kinematics)
+    }
+
+    pub fn mod_spring<F:FnOnce(&mut SpringProperties)>(&mut self, f:F) {
+        f(&mut self.data.borrow_mut().spring)
+    }
+
+    pub fn mod_drag<F:FnOnce(&mut DragProperties)>(&mut self, f:F) {
+        f(&mut self.data.borrow_mut().drag)
+    }
+}
+
+
+
+// ========================
+// === SimulationObject ===
+// ========================
+
+pub trait SimulationObject = HasPosition + 'static;
+
+
+
+// ========================
+// === PhysicsSimulator ===
+// ========================
+
+/// A 60 steps per second physics simulator used to simulate `Properties`.
+pub struct PhysicsSimulator {
+    _fixed_step_animator : FixedStepAnimator
+}
+
+impl PhysicsSimulator {
+    /// Simulates `Properties` on `object`.
+    pub fn new<T:SimulationObject>(mut object:T, mut properties:PhysicsProperties) -> Self {
+        properties.mod_kinematics(|kinematics| { kinematics.set_position(object.position()); });
+
+        let steps_per_second = 60.0;
+        let step_time        = 1.0 / steps_per_second;
+        let current_position = Rc::new(RefCell::new(object.position()));
+        let next_position    = Rc::new(RefCell::new(simulate(&mut properties, step_time)));
+
+        let current_position_clone  = current_position.clone();
+        let next_position_clone     = next_position.clone();
+        let mut continuous_animator = ContinuousTimeAnimator::new(move |time| {
+            let current_position = *current_position_clone.borrow();
+            let next_position    = *next_position_clone.borrow();
+            let transition       = time / step_time / 1000.0;
+            object.set_position(linear_interpolation(current_position, next_position, transition));
+        });
+
+        let _fixed_step_animator   = FixedStepAnimator::new(steps_per_second, move |_| {
+            continuous_animator.set_time(0.0);
+            let mut next_position          = next_position.borrow_mut();
+            *current_position.borrow_mut() = *next_position;
+            *next_position                 = simulate(&mut properties, step_time);
+        });
```

Comment:
I havent seen it before, but why are we using so many naked RefCells here and why do we need to use both `FixedStepAnimator` and `ContinousTimeAnimator` in Physics simulator?

Reply:
@wdanilo

1. `FixedStepAnimator` runs the simulation step.
2. `ContinuousTimeAnimator` runs the simulator every frame, interpolating the current simulation step and next simulation step results.

This can be refactored to only one `ContinuousTimeAnimator`, but it requires the fixed step execution logic to be reimplemented inside `PhysicsSimulator`.

Reply:
We need 1 loop and no re-implementation of logic. I'm 99% sure it would be possible with the right design.